diff --git a/inc/parametrit.inc b/inc/parametrit.inc
index d597a9179..926f86945 100644
--- a/inc/parametrit.inc
+++ b/inc/parametrit.inc
@@ -264,7 +264,7 @@ $pupesoft_port = ($_SERVER['SERVER_PORT'] != '80' and $_SERVER['SERVER_PORT'] !=
 
 // $palvelin muuttujassa on Pupesoft root url, joka tulee salasat.php:stä
 // $palvelin2 muuttujassa on Pupesoft root url, mutta scheme/server/portti on muutettu tarpeen mukaan
-$palvelin2 = $pupesoft_scheme.$pupesoft_server.$pupesoft_port.$pupesoft_url["path"];
+$palvelin2 = $palvelin;
 
 // Haetaan requestin dirname
 $pupesoft_dirname = str_replace(array($phpnimi, $palvelin2), array("", ""), $pupesoft_scheme.$_SERVER["SERVER_NAME"].$_SERVER["SCRIPT_NAME"]);
diff --git a/logout.php b/logout.php
index c64bcb27a..a453ca256 100644
--- a/logout.php
+++ b/logout.php
@@ -6,7 +6,7 @@ require ("inc/parametrit.inc");
 
 $query = "UPDATE kuka SET
           session       = '',
-          kesken        = ''
+          kesken        = 0
           WHERE session = '$session'";
 $result = pupe_query($query);
 
